addons/html_builder/static/src/builder/plugins/builder_overlay/builder_overlay_plugin.js

```diff
@@ -1,34 +1,247 @@
 import { Plugin } from "@html_editor/plugin";
-import { BuilderOverlay } from "./builder_overlay";
+import { throttleForAnimation } from "@web/core/utils/timing";
+import { renderToElement } from "@web/core/utils/render";
+import { getScrollingElement, getScrollingTarget } from "@web/core/utils/scrolling";
+import builderUtils from "../../utils/utils";
 
 export class BuilderOverlayPlugin extends Plugin {
     static id = "builder_overlay";
-    static dependencies = ["selection", "overlay"];
-    static resources = (p) => ({
-        change_selected_toolboxes_listeners: p.openBuilderOverlay.bind(p),
-    });
+    static dependencies = ["selection", "local-overlay"];
+    resources = {
+        change_selected_toolboxes_listeners: this.openBuilderOverlay.bind(this),
+    };
+
+    handleCommand(command) {
+        switch (command) {
+            case "STEP_ADDED":
+                this.update();
+                break;
+        }
+    }
 
     setup() {
-        this.overlay = this.dependencies.overlay.createOverlay(BuilderOverlay, {
-            positionOptions: {
-                position: "center",
-            },
+        this.overlayContainer = this.dependencies.localOverlay.makeLocalOverlay(
+            "builder-overlay-container"
+        );
+        this.overlays = [];
+        this.toolboxes = [];
+
+        this.update = throttleForAnimation(this._update.bind(this));
+
+        // Recompute the overlay when the window is resized.
+        window.addEventListener("resize", this.update);
```

Comment:
In a plugin, you should use this.addDomListener
It will remove the listener when we destroy the editor for you